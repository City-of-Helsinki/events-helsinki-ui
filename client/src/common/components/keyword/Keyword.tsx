import classNames from "classnames";
import React, { FunctionComponent } from "react";

import styles from "./keyword.module.scss";

interface Props {
  color?: "lightEngel50";
  keyword: string;
}

const Keyword: FunctionComponent<Props> = ({ color, keyword }) => {
  return (
    <div className={classNames(styles.keyword, color && styles[color])}>
      {keyword}
    </div>
  );
};

export default Keyword;
