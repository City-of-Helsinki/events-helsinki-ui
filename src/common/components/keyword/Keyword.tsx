import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import styles from './keyword.module.scss';

interface Props {
  blackOnMobile?: boolean;
  color?: 'engelLight50' | 'tramLight20';
  hideOnMobile?: boolean;
  keyword: string;
  onClick: () => void;
}

const Keyword: FunctionComponent<Props> = ({
  blackOnMobile,
  color,
  hideOnMobile,
  keyword,
  onClick,
}) => {
  const handleClick = (ev: React.MouseEvent) => {
    ev.preventDefault();
    onClick();
  };

  return (
    <button
      className={classNames(styles.keyword, color && styles[color], {
        [styles.blackOnMobile]: blackOnMobile,
        [styles.hideOnMobile]: hideOnMobile,
      })}
      onClick={handleClick}
      type="button"
    >
      {keyword}
    </button>
  );
};

export default Keyword;
