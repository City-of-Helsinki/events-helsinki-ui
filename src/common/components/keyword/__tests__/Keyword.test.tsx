import * as React from 'react';
import renderer from 'react-test-renderer';

import Keyword from '../Keyword';

test('Keyword matches snapshot', () => {
  const component = renderer.create(<Keyword keyword="test keyword" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
