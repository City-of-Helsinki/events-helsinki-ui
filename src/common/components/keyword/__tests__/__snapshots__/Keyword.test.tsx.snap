// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  class="keyword"
  role="link"
  type="button"
>
  test keyword
</button>
`;
